### ISSUE TITLE
`QUALIFY` keyword is not recognised for Redshift

### ISSUE THREAD

**Issue by pilou-komoot** [sqlfluff/sqlfluff#5521]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I used `qualify` in a dbt model, SQLFluff is not able to parse it and returns a parsing error.

### Expected Behaviour

I expect the keyword to be recognised as valid.

### Observed Behaviour

`Found unparsable section`

### How to reproduce

```sql
select *
from {{ ref("toto") }}
where 1 = 1
qualify row_number() over () = 1
```

Also note that using `qualify` without a `where` clause requires the `from` clause to be aliased, which conflicts with some SQLFluff rules (see: https://docs.aws.amazon.com/redshift/latest/dg/r_QUALIFY_clause.html)

### Dialect

Redshift

### Version

2.1.2

### Configuration

```
[sqlfluff]
dialect = redshift
templater = dbt
processes = -1
verbose = 1

warn_unused_ignores = True
output_line_length = 120

max_line_length = 120

exclude_rules = ambiguous.column_references, structure.column_order

[sqlfluff:indentation]
template_blocks_indent = False # to keep?

[sqlfluff:layout:type:comma]
line_position = leading

# Everything should be lower case

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = lower

[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = lower

[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = lower

# Other rules

[sqlfluff:rules:aliasing.forbid]
# Avoid table aliases in from clauses and join conditions
# https://docs.sqlfluff.com/en/stable/rules.html#rule-aliasing.forbid
force_enable = True

[sqlfluff:rules:references.consistent]
# References should be consistent in statements with a single table
# https://docs.sqlfluff.com/en/stable/rules.html#rule-references.consistent
force_enable = True
single_table_references = consistent

[sqlfluff:rules:references.keywords]
# Keywords should not be used as identifiers
# https://docs.sqlfluff.com/en/stable/rules.html#rule-references.keywords
unquoted_identifiers_policy = all
quoted_identifiers_policy = all
ignore_words = "timestamp",final,id,

[sqlfluff:rules:structure.subquery]
# By default, don't allow subqueries in from or join clauses
# https://docs.sqlfluff.com/en/stable/rules.html#rule-structure.subquery
forbid_subquery_in = both
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Duplicate as this was fixed in #5002 released in v2.2.0
